### PR TITLE
Reader: integrate sites list into new share popover

### PR DESCRIPTION
**Pull request by bluefuton**
Following on from #17066, this PR integrates the sites list into the new Reader Share popover, and displays the same popover for everybody.

screen shot 2017-08-11 at 15 40 10

### CODE REVIEW

File: client/blocks/reader-share/index.jsx [hunk 1]
```diff
@@ -198,48 +199,42 @@ class ReaderShare extends React.Component {
...
-								</PopoverMenuItem>
-							</ReaderPopoverMenu> ),
+					<ReaderPopoverMenu
+						key="menu"
+						context={ this.refs && this.refs.shareButton }
+						isVisible={ this.state.showingMenu }
+						onClose={ this.closeExternalShareMenu }
+						position={ this.props.position }
+						className="popover reader-share__popover"
+						popoverTitle={ translate( 'Share on' ) }
```
Comment (a8ci18n) [inline client/blocks/reader-share/index.jsx]: Hi! I've found a possible matching string that has already been translated **26** times:
`translate( 'Share on:' )` *ES Score: 11*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/reader/integrate-sites-list-in-new-share-popover)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by bluefuton** [comment]
Now works! A bit of visual tuning to do though:

29167183-b47d0d04-7dc9-11e7-920e-26710a4c1ea4

- [ ] Add top border above search box
- [ ] Fix white fade on right of sites list
- [ ] Condense sites in sites list (reduce margin)

**Comment by blowery** [comment]
when you start a search, should we hide twitter and facebook? or would the jump be too weird?

**Comment by jancavan** [comment]
> when you start a search, should we hide twitter and facebook? or would the jump be too weird?

We shouldn't hide them. Twitter/FB isn't part of the search since they're right above it. But yes, the jump would be weird too.

**Comment by bluefuton** [comment]
One way to find out! I've worked it up quickly in `try/reader/sites-list-in-share-popover-hide-social-share-when-searching`. Give it a spin. 

I find it a bit disconcerting, and sometimes the popover moves position because of the change in size.

**Comment by jancavan** [comment]
Yeah, we shouldn't do that. The search box is intended to search your site list.

**Comment by fraying** [comment]
Yeah, agreed. Facebook and Twitter should stay while filtering your sites.

**Comment by jancavan** [comment]
Fixed remaining visual bits in: https://github.com/Automattic/wp-calypso/pull/17087/commits/524e15ece4004d55999058dbb19885b311f9f24b
screenshot 2017-08-13 22 50 26

I don't think we need the "No sites found" error message here though. It's not helpful. Can we please remove it?
screenshot 2017-08-13 22 52 09

**Comment by blowery** [comment]
Giving this some 👀

**Comment by blowery** [comment]
Seeing some odd focus / selection issues with the keyboard and hover

weird-focus-selection

Not sure we should hold merging for it, but it's pretty odd. I think it's because we have two separate lists, one for the popover items and then the sites list embedded in the third item.

**Comment by jancavan** [comment]
> I think it's because we have two separate lists, one for the popover items and then the sites list embedded in the third item.

Good catch. Yeah, I think that might be why. I'm going to a look at it, but don't exactly know how long it'll take to fix so I would just merge this.

**Comment by fraying** [comment]
>I don't think we need the "No sites found" error message here though. It's not helpful. Can we please remove it?

Agreed!

**Comment by fraying** [comment]
Any, yeah, that hover issue is a major bummer. Any way around it?

File: client/blocks/reader-share/index.jsx [hunk 2]
```diff
@@ -198,51 +193,49 @@ class ReaderShare extends React.Component {
...
+						</PopoverMenuItem>
+						<PopoverMenuItem
+							action="twitter"
+							className="reader-share__popover-item"
+							title={ translate( 'Share on Twitter' ) }
+						>
+							<SocialLogo icon="twitter" />
+							<span>Twitter</span>
+						</PopoverMenuItem>
+						{ this.props.hasSites
```
Comment (samouri) [inline client/blocks/reader-share/index.jsx]: this is a pretty meh way of expressing the conditional render.
lets switch it from:
```jsx
this.props.hasSites ? <SitesSelector ... /> : null
``` 
to
```jsx
this.props.hasSites && <SiteSelector .... />
```

**Comment by jancavan** [comment]
In https://github.com/Automattic/wp-calypso/pull/17087/commits/c59f60aca455c13c9638003072520fddce127638, I updated the arrow color when its on the right of the popover.

**Before:**
screenshot 2017-08-14 13 02 04

**After:**
screenshot 2017-08-14 13 01 37

**Comment by jancavan** [comment]
In https://github.com/Automattic/wp-calypso/pull/17087/commits/017e039978d3d5535760699453748813f6bc6e22:

**Before:**
screenshot 2017-08-14 13 14 53

**After:**
screenshot 2017-08-14 13 16 14

**Comment by jancavan** [comment]
> I don't think we need the "No sites found" error message here though. It's not helpful. Can we please remove it?

For some reason, I'm no longer able to repro this for accounts with zero sites.

But I did notice another odd behavior. When you search for a site and there isn't a match, clicking on the dismiss icon dismisses the entire popover instead of clearing the search. It seems to be the current behavior as well.

Video: https://cloudup.com/c_gYg40DGd6

**Comment by blowery** [comment]
> For some reason, I'm no longer able to repro this for accounts with zero sites.

I fixed that in https://github.com/Automattic/wp-calypso/pull/17087/commits/1b9a7edc722704a7cc56617f2f4872c2719b9135

File: client/blocks/reader-share/index.jsx (hunk 2 above)
Comment (blowery) [inline client/blocks/reader-share/index.jsx]: oh right, react suppresses booleans doesn't it

**Comment by blowery** [comment]
Fixed #17193 in https://github.com/Automattic/wp-calypso/pull/17087/commits/75421e9f607b8ff359c653063864e961ff8afb0e

**Comment by blowery** [comment]
fixed the hover / focus issue by killing granting focus on hover. behaves like you would expect now, except arrows keys no longer work.

File: client/blocks/reader-share/index.jsx (hunk 2 above)
Comment (blowery) [inline client/blocks/reader-share/index.jsx]: fixed

**Comment by jancavan** [comment]
Works great! Confirming it also fixes: https://github.com/Automattic/wp-calypso/issues/17193

**Comment by samouri** [comment]
Code LGTM.  Checking out behavior now

**Comment by bluefuton** [comment]
Thanks for getting this 🚢  everyone!

**Pull request merged**